include/revng/ADT/UpcastablePointer/YAMLTraits.h

```diff
@@ -45,6 +45,10 @@ void dispatchMappingTraits(llvm::yaml::IO &TheIO, O &Obj) {
 template<UpcastablePointerLike T>
 struct PolymorphicMappingTraits {
   static void mapping(llvm::yaml::IO &TheIO, T &Obj) {
+    // Skip empty pointers when serializing
+    if (TheIO.outputting() && Obj.isEmpty())
+      return;
```

Comment:
> could not crash on empty print

The commit message is confusing

Reply:
Whoops, I meant "do not crash". Thanks for catching it.

Reply:
Fixed.